### ISSUE TITLE
UI tests are failing

### ISSUE THREAD

**Issue by timbru** [NLnetLabs/krill#603]
Seemingly unrelated to the actual code changes, all open PRs have UI tests failing. Seems to be a timeout issue somewhere.

**Comment by ximon18** [comment]
**Note:** `cargo test --features ui-tests` passes locally on my dev machine for PR #590 so the UI tests are capable of passing but are not passing on GitHub Actions at present.

Summary of current failures:

| PR | Ubuntu | Rust | Features | Failure |
|---|---|---|---|---|
| #590 | ubuntu-latest | stable | `--features functional-tests,ui-tests` | `multi_user_config_file_with_ta.js`: ESOCKETTIMEOUT on https://localhost:3000/ |
| #591 | ubuntu-latest | beta | `--all-features` | `multi_user_config_file_with_ta.js`: ESOCKETTIMEOUT on https://localhost:3000/ |
| #594 | ubuntu-latest | stable | `--no-default-features` | `cli::client::tests::init_config_file` and `cli::client::tests::init_multi_user_config_file` |
| #596 | ubuntu-latest | 1.47.0 | `--no-default-features` | `error: could not compile 'krill'` |
| #597 | ubuntu-latest | stable | `--all-features` | `multi_user_config_file_with_ta.js`: ESOCKETTIMEOUT on https://localhost:3000/ |

So the issue is independent of Rust version or features used. Interestingly two of the builds failed with `--no-default-features` which precludes running of the UI tets which seems to contradict the issue description, these seem to be different issues. The remaining builds all failed in the same UI test however which seems to be britle on GitHub Actions.

**Comment by ximon18** [comment]
The experiment didn't solve the issue. Checking the current situation there don't appear to be any new types of failure, all CI tests are failing because of the above mentioned ESOCKETTIMEOUT issue. I'll try and look at this again.

**Comment by ximon18** [comment]
Asher several tweaks the tests now seem to be consistently passing on most worrie workflow matrix jobs but always going stuck on one or two with GitHub ringer logs failing to display until the job is canceled.

To me this implies that the GitHub log relaying process on the runner VM is unable to get enough system resources to do its job, i.e. all CPU time  is completely taken by the Krill + OpenID Connect Mock process and the Docker container running Google Chrome the Lagosta UI and the Cypress test runner.

Once canceled the workflow log then shows in most/all? cases that the last logged message was from the OpenID Connect Mock saying it has received the following request (but hadn’t yet answered it??):

`Received Request(GET /authorize?response_type=code&client_id=client-id-123&state=QZV-y22Ks6W4FxH0y4CMWQ&redirect_uri=https%3A%2F%2Flocalhost%3A3000%2Fauth%2Fcallback&scope=openid+email&nonce=VbjvCJslTy38RLZS0F-_eaQFbj6sTWYUTNRHd4XnYQk&prompt=login from 127.0.0.1:33386)`

**Comment by ximon18** [comment]
The hanging issue was due to backward incompatible changes in the Cypress cy.intercept() interface that caused the subsequent wait() to block, presumably there is also an incorrect or missing timeout here as well otherwise it shouldn't have been able to block for hours. This has been fixed now in the dev branch and open PR branches. However, failures remain in the UI tests which need to be investigated and resolved.

**Comment by ximon18** [comment]
Ignoring all unhandled JS promise rejections and commenting out intercepts and waits in the `multi_user_openid_connect_test` test seems to have resolved the remaining issues. I tried using Cypress timeouts with the `wait()` call and disabling browser caching (which is known to break Cypress intercepts) but neither stopped the problematic intercept+wait blocking the test for hours until GH terminates it. Enabling Krill `TRACE` level logging shows that Krill is still running but the Cypress test is blocking on one the awaited intercepts in the test. Enabling full Cypress debug logging didn't yield any obvious insights. For now I have commented out these problematic intercept+wait pairings. The intercepts+waits were introduced to make the test more deterministic so ideally they would be kept but if the tests work in GitHub Actions CI for now without them then that's an improvement over the current situation. The fixes have now been pushed to the `dev` and all open PR branches, hopefully the GH Actions runs now pass.
